### ISSUE TITLE
Duplicates generated with Jekyll 3 and jekyll-sitemap 0.9.0

### ISSUE THREAD

**Issue by mmistakes** [jekyll/jekyll-sitemap#91]
Was doing some testing with Jekyll 3.0.1 locally and noticed a small bug with the `sitemap.xml` being generated is creating duplicates across the board. Not sure if Jekyll 3 is to blame or not.

With Jekyll 2.5.3 and jekyll-sitemap 0.9 I didn't notice the duplicates. For reference here's a [`sitemap.xml`](https://mademistakes.com/sitemap.xml) generated with 2.5.3 and the [same file](https://gist.github.com/mmistakes/de7b3aaea59ada498485) with Jekyll 3.0.1

If you check the first entry you'll notice it shows up again at the end of the file but without `lastmod`. Seems to happen for all documents.

``` html
<url>
    <loc>http://localhost:4000/procreate-paintings/molly-m-portrait/</loc>
    <lastmod>2015-12-07T00:00:00-05:00</lastmod>
</url>
```

``` html
<url>
      <loc>http://localhost:4000/procreate-paintings/molly-m-portrait/</loc>
</url>
```

**Comment by benbalter** [comment]
I suspect posts are being added once as a post, and again as a collection. We'll also want to update the logic to add a last modified date for all collections with dates, not just posts.

**Comment by parkr** [comment]
Fixed by #92.
